### PR TITLE
Process vulnerabilities and map to agents

### PR DESCRIPTION
**Pull request by erkangz**

### CODE REVIEW

File: src/ThreatStackClient.ts [hunk 1]
```diff
@@ -157,14 +182,39 @@ export default class ThreatStackClient {
...
+      const code = err.response.status;
+      if (code === 401) {
+        throw new IntegrationInstanceAuthenticationError(err);
+      } else if (code === 403) {
+        throw new IntegrationInstanceAuthorizationError(err, firstUri);
+      } else if (code === 500) {
+        this.logger.warn(
+          `Server error from ${this.provider} while retrieving ${firstUri}`,
+        );
+        return [];
```
Comment (aiwilliams) [inline src/ThreatStackClient.ts]: Should this be `undefined`? Will this lead to deleting things when a diff is performed?

File: src/util/getCVE.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { PersistedObjectAssignable } from "@jupiterone/jupiter-managed-integration-sdk/jupiter-types";
+
+const cveLink = "https://nvd.nist.gov/vuln/detail/";
+
+export interface CVEProperties {
+  open?: boolean;
```
Comment (fomentia) [inline src/util/getCVE.ts]: CVE entities are supposed to be mapped and global, right (common vulnerabilities and exposures)? If this is mapped, it'll be setting an `open` property on a global, cross-integration entity, which is not good. Shouldn't `open`, `suppressed`, and other state values be on Finding entities?

File: src/ThreatStackClient.ts (hunk 1 above)
Comment (erkangz) [reply src/ThreatStackClient.ts]: ah right.

File: src/util/getCVE.ts (hunk 2 above)
Comment (erkangz) [reply src/util/getCVE.ts]: this util should be moved to the SDK. and this should define a global interface for the CVE.  a CVE can be set to be open or suppressed globally which will have the effect on all related/impacted resources.  I can envision those can be set by the user / security team member via the UI or API.

**Comment by erkangz** [comment]
> If we don't have a `Finding` between the agent and a CVE, and the vulnerability is found, closed, and found again, won't we lose that information?

I don't see a good way around this for the TS integration. We have the same issue even if we add the intermediary `Finding` entity, because closed vulnerabilities in TS are not captured.

**Comment by erkangz** [comment]
@fomentia @aiwilliams I think we should allow for a more flexible pattern and support

- `resource - has -> finding - is -> vuln` OR `resource - has -> vuln`
- and similarly `provider - has -> agent - manages -> device` or `provider - manages -> device`

**Pull request merged**